### PR TITLE
[Front-End] Mobile Interface Done

### PR DESCRIPTION
**Pull request by nex-pixel**
## Feature Description/Implementation
Seperate Routing system with routing file
Changed App and Login page to route to correct url and pass correct router stuff
Added two book look up and book detail mobile pages

## Dependencies
There was a dependency issue that I fixed by correcting react version code of react-qr-barcode-scanner

### CODE REVIEW

File: frontend/src/pages/books/Mobile/BookDetailMobile.tsx [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+              <p className="p-component p-text-secondary text-900 text-xl text-center my-auto">
+                {lastMonthSales}
+              </p>
+            </div>
+          </div>
+        </div>
+      </div>
+      <Divider align="center">
+        <div className="inline-flex align-items-center">
+          <b>Book Related Transactions and Adjustments</b>
```
Comment (caseyszilagyi) [inline frontend/src/pages/books/Mobile/BookDetailMobile.tsx:350]: This should be removed, right?

**Review by caseyszilagyi — CHANGES_REQUESTED** [review]

File: frontend/src/components/text/TextLabels.tsx [hunk 2]
```diff
@@ -7,8 +7,8 @@ export default function TextLabel(props: TextLabelProps) {
   return (
     <label
       className={
-        "p-component p-text-secondary text-teal-900 my-auto mr-2 " +
-        props.labelClassName
+        props.labelClassName ??
+        "p-component p-text-secondary text-teal-900 my-auto mr-2"
```
[the remark this answers is not in the document]
Comment (nex-pixel) [reply frontend/src/components/text/TextLabels.tsx:11]: This does not remove any label properties on any other class as the original method didn't have any labelClassName inputs from any function. The problem was the my-auto was doing odd formatting which is why I had to make it optional.

File: frontend/src/pages/books/Mobile/BookDetailMobile.tsx [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+
+  const calculateDaysOfSupply = (book: Book) => {
+    if (book.stock === 0) {
+      return "(inf)";
+    } else {
+      return Math.floor((book.stock / book.lastMonthSales!) * 30);
+    }
+  };
+
+  const updateShelfSpace = (thickness: number | undefined) => {
```
[the remark this answers is not in the document]
Comment (nex-pixel) [reply frontend/src/pages/books/Mobile/BookDetailMobile.tsx:106]: Got it.

File: frontend/src/pages/books/Mobile/BookDetailMobile.tsx [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+          imageSrc: response.image_url,
+          imageHash: Date.now().toString(),
+        });
+        setNumOfRelatedBooks(book.numRelatedBooks);
+        setRelatedBooks(book.relatedBooks!);
+      })
+      .catch(() => showFailure(toast, "Could not fetch book data"));
+  }, [stock, id]);
+
+  const calculateDaysOfSupply = (book: Book) => {
```
[the remark this answers is not in the document]
Comment (nex-pixel) [reply frontend/src/pages/books/Mobile/BookDetailMobile.tsx:98]: Extracted.

File: frontend/src/pages/books/Mobile/BookDetailMobile.tsx [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+  };
+
+  const updateShelfSpace = (thickness: number | undefined) => {
+    const calcThickness = thickness ? thickness : DEFAULT_THICKNESS;
+    setShelfSpace(
+      Math.round((calcThickness * stock + Number.EPSILON) * 100) / 100
+    );
+  };
+
+  const toast = useRef<Toast>(null);
```
[the remark this answers is not in the document]
Comment (nex-pixel) [reply frontend/src/pages/books/Mobile/BookDetailMobile.tsx:113]: Yes they act like a notification badge on screen somewhat so it works.

File: frontend/src/pages/books/Mobile/BookLookup.tsx
[the remark this answers is not in the document]
Comment (nex-pixel) [reply frontend/src/pages/books/Mobile/BookLookup.tsx:1]: good call. accidentally did it.

File: package.json
[the remark this answers is not in the document]
Comment (nex-pixel) [reply package.json:1]: ok

File: frontend/src/pages/books/Mobile/BookLookup.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    BOOKS_API.getBooks({ isbn_only: true, search: textBox })
+      .then((response) => onAPIResponse(response))
+      .catch(() => showFailure(toast, "Book Search Failed"));
+  };
+
+  const onAPIResponse = (response: GetBooksResp) => {
+    showSuccess(toast, "Message Recieved");
+    if (response.results.length > 0) {
+      const book = APIToInternalBookConversion(response.results[0]);
+      navigate(`${"/books/detail/"}${book.id}`);
```
[the remark this answers is not in the document]
Comment (nex-pixel) [reply frontend/src/pages/books/Mobile/BookLookup.tsx:32]: I think I disagree, check slack for more comments.

File: frontend/src/pages/books/Mobile/BookDetailMobile.tsx (hunk 1 above)
Comment (nex-pixel) [reply frontend/src/pages/books/Mobile/BookDetailMobile.tsx:350]: The mobile interface requirement said the Book detail requirement had to be implemented which is what I did, but we can have professor elaborate more upon it in class.

File: frontend/src/pages/books/Mobile/BookLookup.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (caseyszilagyi) [reply frontend/src/pages/books/Mobile/BookLookup.tsx:36]: Ok, I agree. we can have two separate routers

**Review by caseyszilagyi — APPROVED** [review]

**Pull request merged**